--- a/hald/linux/addons/addon-input.c
+++ b/hald/linux/addons/addon-input.c
@@ -244,10 +241,21 @@
 					}
 				}
 			}
-		} else if (event.type == EV_KEY && key_name[event.code] != NULL && event.value == 1) {
+		} else if (event.type == EV_KEY && key_name[event.code] != NULL) {
+			switch(event.value) {
+				case 0:
+					event_name = "ButtonReleased";
+					break;
+				case 1:
+					event_name = "ButtonPressed";
+					break;
+				case 2:
+					event_name = "ButtonRepeat";
+					break;
+			}
 			dbus_error_init (&error);
-			libhal_device_emit_condition (ctx, udi, 
-						      "ButtonPressed",
+			libhal_device_emit_condition (ctx, udi,
+						      event_name,
 						      key_name[event.code],
 						      &error);
 		}
