diff --git a/hald/linux/addons/addon-input.c b/hald/linux/addons/addon-input.c
index 6f3167f..ba4cb00 100644
--- a/hald/linux/addons/addon-input.c
+++ b/hald/linux/addons/addon-input.c
@@ -167,7 +167,11 @@ static char *key_name[KEY_MAX + 1] = {
 #ifndef KEY_WLAN
 #define KEY_WLAN 238
 #endif
-	[KEY_WLAN] = "wlan"
+	[KEY_WLAN] = "wlan",
+#ifndef KEY_FN
+#define KEY_FN 464
+#endif
+	[KEY_FN] = "fn",
 };
 
 
