--- scripts/linux/hal-system-lcd-set-brightness-linux	2007-05-11 09:29:30.000000000 +0200
+++ scripts/linux/hal-system-lcd-set-brightness-linux	2007-05-15 23:10:58.000000000 +0200
@@ -59,6 +59,9 @@
 	# echo "{0..7} || {0...11}" > /proc/omnibook/lcd
 	# http://bugzilla.gnome.org/show_bug.cgi?id=331458
 	echo "$value" > $HAL_PROP_LINUX_ACPI_PATH
+elif [ "$HAL_PROP_LAPTOP_PANEL_ACCESS_METHOD" = "fjtbl" ]; then
+	# echo "{1..8}" > /proc/acpi/video/VGA/LCD/brightness
+	echo "$value" > /proc/acpi/video/VGA/LCD/brightness
 else
 	echo "org.freedesktop.Hal.Device.LaptopPanel.NotSupported" >&2
 	echo "No ACPI method found" >&2
